app/controllers/finders_controller.rb

```diff
@@ -84,4 +85,8 @@ def result_set_presenter_class
   def org_registry
     @org_registry ||= Registries::OrganisationsRegistry.new
   end
+
+  def parent
+    params.fetch(:parent, '')
```

Comment:
I'm not 100% clear on how and where the params are sanitised but if they aren't then should this not clean the values before being used? I know the field is not visible to the user but that wouldn't stop a hacker from attempting to pass in any value they like.

Reply:
@aliuk2012 you're right that a user can provide any value they like for the parent param. Could you provide an example of how this param might be abused?

The parent param is already used for [looking up an organisation in a Hash](https://github.com/alphagov/finder-frontend/pull/879/files/ac7f87e44d48b98b765d173dbe787c454eada966#diff-eb2d1de38a399388bf54245e59b8dfc7R68) and with this change we also render the value as the value of a hidden field.

We don't use this param for any queries or in any other way so we don't need to call `ParamsCleaner` on it. In this case the biggest threat is probably that the hacker manipulates the DOM so the user gets a different experience than intended.

If we try some XSS by setting the parent param as `<img src=x\\x11onerror=\"javascript:alert(1)\">`, this is rendered in an ERB template, and eventually ends up as this in this DOM:

```
<input type="hidden" name="parent" value="<img src=x\\x11onerror=\&quot;javascript:alert(1)\&quot;>">
```

Because ERB escapes strings when we do things like `value="<%= @parent %>`, this handles the majority of issues like XSS.

https://finder-frontend-pr-879.herokuapp.com/news-and-communications?parent=%3Cimg%20src=x\\x11onerror=\%22javascript:alert(1)\%22%3E.

Reply:
I don't have an example of what might be passed through but if its not used in any query or in other way then it shouldn't be a problem. Also it's also good to know that ERB provides some security by escaping strings